velero-plugins/pod/secrets.go

```diff
@@ -0,0 +1,56 @@
+package pod
+
+import (
+	"strings"
+
+	"github.com/openshift/library-go/pkg/build/naming"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// This is a copy of the function `needsDockercfgSecret` and its dependencies from the OpenShift to decide if we need to wait for the docker secret to be created
+
+// We follow `func needsDockercfgSecret(serviceAccount *v1.ServiceAccount) bool {` logic
+// to decide if we need to wait for the secret to be created where the service account checked is "default"
+// https://github.com/openshift/openshift-controller-manager/blob/master/pkg/serviceaccounts/controllers/create_dockercfg_secrets.go#L304
+
+const (
+	// These constants are here to create a name that is short enough to survive chopping by generate name
+	maxNameLength             = 63
+	randomLength              = 5
+	maxSecretPrefixNameLength = maxNameLength - randomLength
+)
+
+func needsDockercfgSecret(serviceAccount *v1.ServiceAccount) bool {
+	mountableDockercfgSecrets, imageDockercfgPullSecrets := getGeneratedDockercfgSecretNames(serviceAccount)
```

Comment:
Upon testing, I think this may not be reliable indicator as it's "racy" for one... as SA.secret can eventually show up from the controller code that reconcile the SA on new namespace creation.

I believe the original intent of this [function](https://github.com/openshift/openshift-controller-manager/blob/master/pkg/serviceaccounts/controllers/create_dockercfg_secrets.go#L304) was to add secrets to SA if field is already not populated, rather than to detect if secret needs to be created.

So we cannot use this method. May have to go back to reading cluster capability.

Reply:
Without reading into other namespaces, an alternative but slow is at 5 minutes, we assume no secret will be created and so ImageRegistry capability is assumed to be disabled.